### ISSUE TITLE
get column number

### ISSUE THREAD

**Issue by METOUIA7** [jacobslusser/ScintillaNET#75]
hi jacob
i want to know how to get the number of colmn in line x
for exapmle if the cursor in line 12 column 3 
how to get column 3

**Comment by jacobslusser** [comment]
TL;DR
The `GetColumn` method.

That's an interesting question when you think about it... We could use the `CurrentPosition` property to get the position of the caret within the document; use that with `LineFromPosition` to get the line index; get the line start position from `Line.Position` and then finally subtract that from the original caret position to get the "column":

```cs
var currentPos = scintilla.CurrentPosition;
var currentLine = scintilla.LineFromPosition(currentPos);
var linePos = scintilla.Lines[currentLine].Position;
var fauxColumn = (currentPos - linePos);
```

This wouldn't be the "column" though (hence my `fauxColumn` variable name). Why? Because it doesn't account for tab characters which can occupy 4, 8, or any other amount of spaces. Usually when someone says they want the "column", they want the line space as it would show on a ruler, which is not necessarily the number of characters. To get a column position--while also taking into account the tab size--use the `GetColumn` method:

```cs
var column = scintilla.GetColumn(scintilla.CurrentPosition);
```

**Comment by METOUIA7** [comment]
thanks alote for help it's work good
